baybe/objective.py

```diff
@@ -1,155 +1,38 @@
-"""Functionality for defining optimization objectives."""
+"""Temporary dispatching functionality for backward compatibility."""
 
-from __future__ import annotations
+import warnings
+from typing import Any, Literal, Optional
 
-from functools import partial
-from typing import Any, Literal
-
-import numpy as np
-import pandas as pd
-from attr import define, field
-from attr.validators import deep_iterable, in_, instance_of, min_len
-
-from baybe.serialization import SerialMixin
+from baybe.objectives.desirability import DesirabilityObjective
+from baybe.objectives.single import SingleTargetObjective
 from baybe.targets.base import Target
-from baybe.targets.numerical import NumericalTarget
-from baybe.utils.numerical import geom_mean
-
-
-def _normalize_weights(weights: list[float]) -> list[float]:
-    """Normalize a collection of weights such that they sum to 100.
-
-    Args:
-        weights: The un-normalized weights.
-
-    Returns:
-        The normalized weights.
-    """
-    return (100 * np.asarray(weights) / np.sum(weights)).tolist()
-
 
-@define(frozen=True)
-class Objective(SerialMixin):
-    """Class for managing optimization objectives."""
 
-    # TODO: The class currently directly depends on `NumericalTarget`. Once this
-    #   direct dependence is replaced with a dependence on `Target`, the type
-    #   annotations should be changed.
-
-    mode: Literal["SINGLE", "DESIRABILITY"] = field()
-    """The optimization mode."""
-
-    targets: list[Target] = field(validator=min_len(1))
-    """The list of targets used for the objective."""
-
-    weights: list[float] = field(converter=_normalize_weights)
-    """The weights used to balance the different targets. By default, all
-    weights are equally important."""
-
-    combine_func: Literal["MEAN", "GEOM_MEAN"] = field(
-        default="GEOM_MEAN", validator=in_(["MEAN", "GEOM_MEAN"])
+def Objective(
+    mode: Literal["SINGLE", "DESIRABILITY"],
+    targets: list[Target],
+    weights: Optional[list[float]] = None,
+    combine_func: Optional[Literal["MEAN", "GEOM_MEAN"]] = None,
+):
+    """Return the appropriate new-style class depending on the mode."""
+    warnings.warn(
+        "The use of `baybe.objective` is deprecated and will be disabled in "
+        "a future version. Use the classes defined in `baybe.objectives` instead.",
+        DeprecationWarning,
     )
-    """The function used to combine the different targets."""
-
-    def __str__(self) -> str:
```

Comment:
The `__str__` method for the new classes is not yet there. How shall we go about this?

Reply:
Well.... we should write it/them? Is there any argument not to?

Reply:
Yeah, sure, my question was more: will @RimRihana take care of it, potentially in a mini follow-up PR?

Reply:
Ok, I've done so, but only by reintroducing the old stuff with the necessary minor modifications to make it work with the new classes. While doing so, I still noticed some problems with the `__str__` methods but fixing these will have to wait until we are allowed to do maintenance PRs again 😄

Reply:
Would you mind to briefly explain the problems that you encountered? Even if we do not fix them now, we then have them written down somewhere already (and we will for sure remember that....)